### PR TITLE
Don't remove config files with podman system reset

### LINKED ISSUE #7447: podman system reset blows away storage configuration file
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**

1. Set up a storage configuration file (`~/.config/containers/storage.conf`) -- for example to set up `overlay`

2. Run `podman system reset`

**Describe the results you received:**

```console
$ podman system reset

WARNING! This will remove:
        - all containers
        - all pods
        - all images
        - all build cache
Are you sure you want to continue? [y/N] y
```

No mention of blowing away configuration (I really don't expect this command to destroy configuration either), and yet `~/.config/containers/storage.conf` is gone.

**Describe the results you expected:**

Either improved messaging that the configuration will be deleted (if this is intentional) or to not delete the configuration file.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```console
$ podman version
Version:      2.0.4
API Version:  1
Go Version:   go1.14.2
Built:        Wed Dec 31 16:00:00 1969
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```console
$ podman info --debug
host:
  arch: amd64
  buildahVersion: 1.15.0
  cgroupVersion: v1
  conmon:
    package: 'conmon: /usr/libexec/podman/conmon'
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.20, commit: '
  cpus: 5
  distribution:
    distribution: ubuntu
    version: "20.04"
  eventLogger: file
  hostname: babibox
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.4.0-42-generic
  linkmode: dynamic
  memFree: 6446026752
  memTotal: 8348299264
  ociRuntime:
    name: runc
    package: 'containerd.io: /usr/bin/runc'
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc10
      commit: dc9208a3303feef5b3839f4323d9beb36df0a9dd
      spec: 1.0.1-dev
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: 'slirp4netns: /usr/bin/slirp4netns'
    version: |-
      slirp4netns version 1.1.4
      commit: unknown
      libslirp: 4.2.0
      SLIRP_CONFIG_VERSION_MAX: 2
  swapFree: 1964396544
  swapTotal: 1964396544
  uptime: 10m 5.43s
registries:
  search:
  - docker.io
  - quay.io
store:
  configFile: /home/asottile/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: 'fuse-overlayfs: /usr/bin/fuse-overlayfs'
      Version: |-
        fusermount3 version: 3.9.0
        fuse-overlayfs: version 1.1.0
        FUSE library version 3.9.0
        using FUSE kernel interface version 7.31
  graphRoot: /home/asottile/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 0
  runRoot: /run/user/1000/containers
  volumePath: /home/asottile/.local/share/containers/storage/volumes
version:
  APIVersion: 1
  Built: 0
  BuiltTime: Wed Dec 31 16:00:00 1969
  GitCommit: ""
  GoVersion: go1.14.2
  OsArch: linux/amd64
  Version: 2.0.4
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```console
$ apt list podman
Listing... Done
podman/unknown,now 2.0.4~1 amd64 [installed]
podman/unknown 2.0.4~1 arm64
podman/unknown 2.0.4~1 armhf
podman/unknown 2.0.4~1 s390x
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

virtualbox, but also aws

### PR DESCRIPTION
**Pull request by Luap99**
Check if storage.conf exists and display a message that
this file should be removed if it has not been modified.

Fixes #7447

### CODE REVIEW

**Comment by Luap99** [comment]
@mheon Is this satisfactory for you?

**Comment by mheon** [comment]
Sure, this should be fine

File: libpod/reset.go [hunk 1]
```diff
@@ -104,8 +106,12 @@ func (r *Runtime) Reset(ctx context.Context) error {
 	}
 
 	if rootless.IsRootless() {
-		configPath := filepath.Join(os.Getenv("HOME"), ".config/containers")
-		if err := os.RemoveAll(configPath); err != nil {
+		if storageConfPath, err := storage.DefaultConfigFile(true); err == nil {
```
Comment (rhatdan) [inline libpod/reset.go:109]: This same check should be done as root, to see if the /etc/containers/storage.conf file has been created.

**Comment by rhatdan** [comment]
LGTM

**Comment by mheon** [comment]
@Luap99 Can you rebase? the rootless RCLI:true test is once again passing on master

**Comment by TomSweeneyRedHat** [comment]
LGTM
but it looks like the endpoint test might be misbehaving now.

**Pull request merged**